### PR TITLE
Feat: Query Publisher Sequence

### PR DESCRIPTION
**Pull request by kuluk1**
Adds request and response for query publisher sequence

https://github.com/coders51/rabbitmq-stream-js-client/issues/15

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
     "check-ts": "tsc --noEmit && tsc --project test/tsconfig.json --noEmit",
     "check-lint": "tsc --noEmit && eslint 'src/**/*.ts' 'test/**/*.ts'",
     "check-format": "prettier -c 'src/**/*.ts' 'test/**/*.ts'",
-    "check-spell": "cspell 'src/**/*ts' 'test/**/*ts'"
+    "check-spell": "cspell 'src/**/*ts' 'test/**/*ts'",
+    "format": "prettier -w './**/*.{ts,md,js,json}'"
```
Comment (gpad) [inline package.json:14]: If you want to add the `format` command you should be consistent with the `check-format`, we verify only `ts`

File: src/connection.ts [hunk 2]
```diff
@@ -115,6 +117,45 @@ export class Connection {
...
+    if (!res.ok) {
+      throw new Error(`Delete Stream command returned error with code ${res.code}`)
+    }
+    this.logger.debug(`Delete Stream response: ${res.ok} - '${inspect(params.stream)}'`)
+    return res.ok
+  }
+
+  public async queryPublisherSequence(params: { stream: string; publisherRef: string }): Promise<bigint> {
+    const res = await this.sendAndWait<QueryPublisherResponse>(new QueryPublisherRequest(params))
+
```
Comment (gpad) [inline src/connection.ts:146]: Try to avoid empty lines ... (I know it0s my OCD ...)

File: src/producer.ts [hunk 3]
```diff
@@ -9,4 +9,8 @@ export class Producer {
       new PublishRequest({ publisherId: this.publisherId, messages: [{ publishingId, message }] })
     )
   }
+
+  getLastPublishingId(stream: string, publisherRef: string) {
```
Comment (gpad) [inline src/producer.ts:13]: Why don't you add `stream` and `publisherRef` to the ctor of Producer? These two value is related to producer

File: test/e2e/query_publisher_sequence.test.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    await p1.send(2n, Buffer.from(`test${randomUUID()}`))
+    await p1.send(3n, Buffer.from(`test${randomUUID()}`))
+    await p1.send(4n, Buffer.from(`test${randomUUID()}`))
+
+    const lastPublishingId = await p2.getLastPublishingId(testStreamName, publisherRef)
+
+    expect(lastPublishingId).to.be.equal(4n)
+
+    await connection.close()
+  }).timeout(10000)
```
Comment (gpad) [inline test/e2e/query_publisher_sequence.test.ts:44]: Keep the `AAA` structure

File: package.json (hunk 1 above)
Comment (kuluk1) [reply package.json]: left only `ts` for the format

File: src/connection.ts (hunk 2 above)
Comment (kuluk1) [reply src/connection.ts]: I'll keep in mind, removed the empty line

File: src/producer.ts (hunk 3 above)
Comment (kuluk1) [reply src/producer.ts]: added the values

File: test/e2e/query_publisher_sequence.test.ts (hunk 4 above)
Comment (kuluk1) [reply test/e2e/query_publisher_sequence.test.ts]: moved lines around

File: test/e2e/query_publisher_sequence.test.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+  beforeEach(async () => {
+    await rabbit.createStream(testStreamName)
+  })
+
+  afterEach(async () => {
+    await rabbit.deleteStream(testStreamName)
+  })
+
+  it("asking for the last sequence read from a publisher returns the last sequence id", async () => {
```
Comment (icappello) [inline test/e2e/query_publisher_sequence.test.ts:19]: Could you add a test about asking the sequence when nothing has been published?

Comment (kuluk1) [reply test/e2e/query_publisher_sequence.test.ts:19]: added the test

**Review by gpad — APPROVED** [review]

**Pull request merged**